### PR TITLE
CSV Output

### PR DESCRIPTION
**Pull request by omerbenamram**
support csv output, and fix a whole bunch of things.

### CODE REVIEW

File: src/entry.rs [hunk 1]
```diff
@@ -215,81 +203,77 @@ impl MftEntry {
...
-                                    }))
-                                }
-                            },
-                            ResidentialHeader::NonResident(_) => Some(Ok(Attribute {
-                                header,
-                                data: MftAttributeContent::None,
-                            })),
-                        }
+
+                let header = header.unwrap();
```
Comment (ohadravid) [inline src/entry.rs]: Use match!

File: src/mft.rs [hunk 2]
```diff
@@ -86,19 +87,26 @@ impl<T: ReadSeek> MftParser<T> {
             if count == total_entries {
                 None
             } else {
+                let entry = Some(self.get_entry(count));
```
Comment (ohadravid) [inline src/mft.rs]: I think a nicer way to do this is `0..total_entries.map(|i| self.get_entry(i))`

File: src/entry.rs (hunk 1 above)
Comment (omerbenamram) [reply src/entry.rs]: It gets really nested and hard to follow (it was using match originally).

File: src/bin/mft_dump.rs [hunk 3]
```diff
@@ -36,22 +86,49 @@ impl MftDump {
...
+            _ => None,
+        };
+
+        let number_of_entries = mft_handler.get_entry_count();
+
+        let chunk_size = 1000;
+        let mut chunk_count = 0;
+        let mut entry_count = 0;
+
+        while entry_count <= number_of_entries {
```
[the remark this answers is not in the document]
Comment (omerbenamram) [reply src/bin/mft_dump.rs]: Because `print_csv_entry` (and basically the whole `get_full_name` thing) borrow the parser mutably (to possibly fetch more records).

(json should also do that, but it's problematic because of the behavior you've mentioned earlier where we use rely on the struct, but we don't want\it's not convenient to always fetch `full_name`)

File: src/attribute/raw.rs [hunk 4]
```diff
@@ -1,15 +1,37 @@
...
-impl ser::Serialize for RawAttribute {
-    fn serialize<S>(&self, serializer: S) -> ::std::result::Result<S::Ok, S::Error>
-    where
-        S: ser::Serializer,
-    {
-        serializer.serialize_str(&utils::to_hex_string(&self.0).to_string())
+#[derive(Serialize, Clone, Debug)]
+pub struct RawAttribute {
+    pub attribute_type: MftAttributeType,
+    #[serde(serialize_with = "data_as_hex")]
```
[the remark this answers is not in the document]
Comment (omerbenamram) [reply src/attribute/raw.rs]: I agree, but I'm not sure how it would work with the way serde works.

File: src/mft.rs (hunk 2 above)
Comment (omerbenamram) [reply src/mft.rs]: nice!!
